Header Fixes #406rda #469xmw

mahamzayed opened the pull request:
# Description

The purpose of this PR is to address the following fixes for the header:
1. Direction of the search icon
2. Overlay and disable scroll on mobile navigation and search
3. Adding a new navigation link for the Atlas page
4. Scrolling on overflow for the mobile navigation when the length of the screen gets too small.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Verify that the four fixes that are listed in the description are now working as expected.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests that prove my fix is effective or that my feature works

mahamzayed commented:
@cameronbaney I didn't add the container class to the header as we discussed previously because @jwagenaar wanted the search bar and the logo to be on the left and right ends of the header, as per the designs. By using the main container class, this would have constrained the width and added margins on both sides of the header. I've reverted back to the fluid width solution again.

Also spoke with Annelise and we decided that anything below 1024px will get the mobile menu. I've made those changes as well.

cameronbaney commented on components/header/Header.vue:
```diff
@@ -172,10 +199,26 @@ export default {
+     * Watches scrollingState in vuex to disable scrolling
+     * on body based on mobile header functionality
+     */
+    scrollingState: {
```
Modifying the body style negates the use of Vuex. This could be done when opening the mobile menu instead. I would recommend adding a class to `layouts/default.vue` that is computed by this Vuex state.

Not using a class also doesn't give us the ability to use media queries for this, which is needed to prevent this from happening:
![localhost_3000_data_searchType=datasets searchTerms=(Laptop with HiDPI screen)](https://user-images.githubusercontent.com/1493195/73085050-123c8d80-3e9c-11ea-9b5a-ff5e02d9335e.png)

Steps to reproduce:
- Make viewport small enough to show the mobile header
- Open mobile menu
- Make viewport large enough to show desktop header

mahamzayed replied:
Great suggestion. I don't know why I didn't think about using Vuex in layouts 🤦‍♀ I'll make the change.

mahamzayed replied:
@cameronbaney made fixes as requested

cameronbaney commented on layouts/default.vue:
```diff
@@ -9,15 +9,25 @@
 
 <style lang="scss" scoped>
 @import '@/assets/_variables.scss';
+.layout {
```
This needs to be in a media query to prevent it from messed up on large viewports, as the original problem is still happening.

Steps to reproduce

1. Make viewport small enough to show the mobile header
2. Open mobile menu
3. Make viewport large enough to show desktop header

mahamzayed replied:
So I was aware of this happening, but I was thinking about it and isn't it not possible for a user to switch to larger viewpoirts on a mobile device? i.e. if a user were on a mobile device and then were to switch to a desktop computer to view the site this issue would no longer be happening. I can add the media query but just a thought.

cameronbaney replied:
This is true, but we should not code for devices, we should code for viewport sizes. This way, we use the mindset of "Does it look good at this size?" rather than "Does it look good on this device?" This also allows us to subscribe to the idea of responsive design for when your design breaks, and not just for N number of sizes. We don't have to worry about the latter on this site as much, but I've have about 13 media queries for one page for some of my other sites in the past because we coded to when the design broke at a certain viewport size.

We cannot control what size the user has their browser on desktop, and having the user look at it on multiple sizes (while resizing) or at smaller sizes on desktop is a possibility.

cameronbaney commented:
How is this progressing?

mahamzayed commented:
> How is this progressing?

I'm going to address your feedback now, sorry have been busy with the project details page all day.

mahamzayed commented:
@cameronbaney Addressed feedback. I was originally going to create modules for vuex but that was proving more complicated than I thought so I kept it simple with `index.js` for now. Let me know if you run into anymore issues.
